sync/class.jetpack-sync-module-posts.php

```diff
@@ -204,14 +203,16 @@ function filter_post_content_and_add_links( $post_object ) {
 	}
 
 	public function save_published( $new_status, $old_status, $post ) {
+
 		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
-			$this->just_published = $post->ID;
+			$this->just_published[] = $post->ID;
 		}
 	}
 
 	public function send_published( $post_ID, $post, $update ) {
-		if ( $this->just_published === $post->ID ) {
-			$this->just_published = null;
+		$found_key = array_search( $post->ID, $this->just_published );
```

Comment:
Because we want to remove the key anyways right after. So having a key works well here.